jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java

```diff
@@ -489,4 +496,58 @@ public String getLogTimeZone()
     {
         return _logTimeZone;
     }
+    
+
+    private String extractUser(Request request)
+    {
+        Authentication authentication = request.getAuthentication();
+        if (authentication instanceof Authentication.User)
+        {
+            return ((Authentication.User) authentication).getUserIdentity().getUserPrincipal().getName();
+        }
+
+        String authorization = request.getHeader(AUTHORIZATION_HEADER);
+        if (authorization == null)
+        {
+            return null;
+        }
+
+        int separator = authorization.indexOf(' ');
+        if (separator < 0)
+        {
+            LOG.debug("No separator in th the authorization http header field was found.");
+            return null;
+        }
+
+        String type = authorization.substring(0, separator);
+        String remaining = authorization.substring(separator + 1);
+        switch (type)
+        {
+            case BASIC_AUTHENTICATION:
+                return extractUsingBasicAuthentication(remaining);
+            default:
+                LOG.debug("Not supported type " + type);
+                return null;
+        }
+    }
+
+    private String extractUsingBasicAuthentication(String remaining) {
+        Base64.Decoder decoder = Base64.getDecoder();
+        String usernamePassword = new String(decoder.decode(remaining), StandardCharsets.UTF_8);
+
+        if (usernamePassword==null || usernamePassword.isEmpty()) {
+            LOG.debug("No username password was specified or it was empty");
+            return null;
+        }
+
+        String[] tokens = usernamePassword.split(":");
+        if (tokens.length!=2) {
+            LOG.debug("Expected only two tokens in in the authorization field but it was "
```

Comment:
I fixed that and added some more context information when authentication type is not supported